nominal/core/client.py

```diff
@@ -212,6 +222,77 @@ def __repr__(self) -> str:
         out += ">"
         return out
 
+    def _workspace_rid_for_search(self, workspace: WorkspaceSearchT) -> str | None:
+        """Provide the correct workspace rid to use when searching (potentially using a provided workspace)
+
+        Args:
+            workspace: Workspace (or None) that user wants to search with
+
+        Returns:
+            If no workspace is provided, then return the default workspace (or none if none is configured).
+            If a workspace is provided, then return it if authenticated, otherwise, return None.
+        """
+        search_rid = None
+        if isinstance(workspace, Workspace):
+            search_rid = workspace.rid
+        elif isinstance(workspace, str):
+            search_rid = workspace
+        elif workspace == WorkspaceSearchType.ALL:
+            return None
+        elif workspace is WorkspaceSearchType.DEFAULT:
+            search_rid = None
+        else:
+            raise ValueError(f"Unexpected workspace: {workspace}")
+
+        try:
+            # NOTE: raises a conjure exception if the given rid is not visible to the user (or doesn't exist period)
+            resolved_workspace = self.get_workspace(search_rid)
+            return resolved_workspace.rid
+        except NominalConfigError:
+            # re-raising with a more specific exception message
+            raise NominalConfigError(
+                "WorkspaceSearchType.DEFAULT provided for workspace rid, but no default configured. "
+                "Specify a workspace_rid within your config profile (see `nom config profile --help`), "
+                "specify a workspace_rid manually, or contact your Nominal representative to set a default "
```

Comment:
```suggestion
                "specify a workspace RID manually, or contact your Nominal representative to set a default "
```
(since the argument isn't `workspace_rid` anymore)